lib/Conversion/ExportVerilog/LegalizeNames.cpp

```diff
@@ -169,8 +169,9 @@ static void legalizeModuleLocalNames(HWModuleOp module,
       if (auto name = op->getAttrOfType<StringAttr>(verilogNameAttr)) {
         nameResolver.insertUsedName(
             op->getAttrOfType<StringAttr>(verilogNameAttr));
-      } else if (isa<sv::WireOp, RegOp, LogicOp, LocalParamOp, hw::InstanceOp,
-                     sv::InterfaceInstanceOp, sv::GenerateOp>(op)) {
+      } else if (isa<sv::WireOp, hw::WireOp, RegOp, LogicOp, LocalParamOp,
+                     hw::InstanceOp, sv::InterfaceInstanceOp, sv::GenerateOp>(
```

Comment:
Not specific to this PR, but I've always felt we should have at least a trait, or maybe an op interface for this collection of ops. Sorry, I can't help but bring it up whenever I look at this.

Reply:
In fact I opened this issue: https://github.com/llvm/circt/issues/2138, but closed it since what I wanted back then was different.

Reply:
Yeah something like a `NameableDeclOpInterface` 😁